Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp

```diff
@@ -74,44 +74,46 @@ namespace MaterialEditor
+        m_entityContext = AZStd::make_unique<AzFramework::EntityContext>();
+        m_entityContext->InitContext();
 
         // Create and register a scene with all available feature processors
         AZ::RPI::SceneDescriptor sceneDesc;
-        sceneDesc.m_nameId = AZ::Name("MaterialViewport");
+        sceneDesc.m_nameId = AZ::Name(AZStd::string::format("MaterialViewportWidgetScene_%i", GetViewportContext()->GetId()));
         m_scene = AZ::RPI::Scene::CreateScene(sceneDesc);
         m_scene->EnableAllFeatureProcessors();
 
         // Bind m_defaultScene to the GameEntityContext's AzFramework::Scene
         auto sceneSystem = AzFramework::SceneSystemInterface::Get();
         AZ_Assert(sceneSystem, "MaterialViewportWidget was unable to get the scene system during construction.");
-        AZStd::shared_ptr<AzFramework::Scene> mainScene = sceneSystem->GetScene(AzFramework::Scene::MainSceneName);
 
-        // This should never happen unless scene creation has changed.
-        AZ_Assert(mainScene, "Main scenes missing during system component initialization");
-        mainScene->SetSubsystem(m_scene);
+        AZ::Outcome<AZStd::shared_ptr<AzFramework::Scene>, AZStd::string> createSceneOutcome =
+            sceneSystem->CreateScene(AZStd::string::format("MaterialViewportWidgetScene_%i", GetViewportContext()->GetId()));
+        AZ_Assert(createSceneOutcome, createSceneOutcome.GetError().c_str());
+        m_frameworkScene = createSceneOutcome.TakeValue();
+        m_frameworkScene->SetSubsystem(m_scene);
+        m_frameworkScene->SetSubsystem(m_entityContext.get());
 
         // Load the render pipeline asset
         AZ::Data::Asset<AZ::RPI::AnyAsset> pipelineAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
             m_defaultPipelineAssetPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::Error);
 
         // The default pipeline determines the initial MSAA state for the application
-        const AZ::RPI::RenderPipelineDescriptor* renderPipelineDescriptor = AZ::RPI::GetDataFromAnyAsset<AZ::RPI::RenderPipelineDescriptor>(pipelineAsset);
+        const AZ::RPI::RenderPipelineDescriptor* renderPipelineDescriptor =
+            AZ::RPI::GetDataFromAnyAsset<AZ::RPI::RenderPipelineDescriptor>(pipelineAsset);
         AZ::RPI::RPISystemInterface::Get()->SetApplicationMultisampleState(renderPipelineDescriptor->m_renderSettings.m_multisampleState);
 
         // Create a render pipeline from the specified asset for the window context and add the pipeline to the scene
-        m_renderPipeline = AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(pipelineAsset, *GetViewportContext()->GetWindowContext().get());
+        m_renderPipeline =
+            AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(pipelineAsset, *GetViewportContext()->GetWindowContext().get());
         pipelineAsset.Release();
         m_scene->AddRenderPipeline(m_renderPipeline);
```

Comment:
I've tried this code on how to handle the render pipeline and it crashes when there are 2 viewports opened.

There are 2 issues:
- Setting the multisample state to the whole application in 2 viewports leads to assert crash:
````
 E:/Amazon/GitHub/o3de/Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp(161): (56980) 'void __cdecl AZ::RPI::RasterPass::UpdateDrawList(void)'
System: View's DrawListTags out of sync with pass'. 
````

- The render pipeline name needs to be unique too, currently it's getting the following error:
````
[Error] (FrameScheduler) - Scope Root.MainPipeline.MainPipeline.MorphTargetPass already exists. You must specify a unique name.
````

I've used the following code and it worked:

````
AZ::RPI::RenderPipelineDescriptor renderPipelineDesc;
{
        // Load the render pipeline asset
        AZ::Data::Asset<AZ::RPI::AnyAsset> pipelineAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
            m_defaultPipelineAssetPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::Error);
        const AZ::RPI::RenderPipelineDescriptor* assetPipelineDesc = AZ::RPI::GetDataFromAnyAsset<AZ::RPI::RenderPipelineDescriptor>(pipelineAsset);
        AZ_Assert(assetPipelineDesc, "Invalid render pipeline descriptor from asset %s", m_defaultPipelineAssetPath.c_str());

        // Use a unique render pipeline name to not conflict with other render pipelines
        renderPipelineDesc = *assetPipelineDesc;
        renderPipelineDesc.m_name += AZStd::string::format("_%i", GetViewportContext()->GetId()); 

        pipelineAsset.Release();
}

m_renderPipeline = AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(renderPipelineDesc, *GetViewportContext()->GetWindowContext().get());
m_scene->AddRenderPipeline(m_renderPipeline);
````

Reply:
Fixed the same render pipeline issue in Animation Editor and UI Editor: https://github.com/o3de/o3de/pull/7751